File: 03-1-login_kt2/app/src/main/java/com/example/login_kt2/repository/room/AppDatabase.kt
package com.example.login_kt2.repository.room

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase

@Database(entities = [LoginTry::class], version = 1, exportSchema = false)
abstract class AppDatabase : RoomDatabase() {

    abstract fun loginTryDao(): LoginTryDao

    companion object {

        @Volatile
        private var Instance: AppDatabase? = null

        fun getDatabase(context: Context): AppDatabase {
            return Instance ?: synchronized(this) {
                Room.databaseBuilder(context, AppDatabase::class.java, "app_database")
                    .build()
                    .also { Instance = it }
            }
        }
    }
}

File: 03-1-login_fl/lib/repository/floor/app_database.dart
// imports needed for the floor classes
import 'package:floor/floor.dart';
import 'dart:async';
import 'package:sqflite/sqflite.dart' as sqflite;

import 'package:login_fl/repository/floor/login_try_dao.dart';
import 'package:login_fl/repository/floor/login_try.dart';

part 'app_database.g.dart';

@Database(version: 1, entities: [LoginTry])
abstract class AppDatabase extends FloorDatabase {

  LoginTryDao get loginTryDao;

}